### ISSUE TITLE
Update windows projects for compatibility with VS 2019 (and new VS 2017 installs)

### ISSUE THREAD

**Issue by stuartmorgan** [google/flutter-desktop-embedding#342]
Per https://github.com/google/flutter-desktop-embedding/issues/338#issuecomment-485255745 VS 2019 complains about the SDK and toolchain versions. We need to determine whether there are any compilation issues after doing the project upgrades.

@clarkezone I'm not familiar with how VS versioning is handled. Is it expected that someone using a newer version would always need to do manual project upgrades, or is there a way to configure the project to allow using x-or-newer component versions rather than exactly-x?

**Comment by MichaelPriebe** [comment]
2019 is working for me, I just had to go in the installer and install `Windows 10 SDK (10.0.17134.0)` it seems `Windows 10 SDK (10.0.17763.0)` (newer version) is installed automatically.

**Comment by MichaelPriebe** [comment]
![image](https://user-images.githubusercontent.com/5238952/56475402-ca449200-6455-11e9-811b-6e17d75e466d.png)

**Comment by Saifallak** [comment]
@MichaelPriebe
You're using VS2017 (v141) C++ build tools, and that's what i said flutter desktop doesn't use VS2019 (v142) build tools

**Comment by clarkezone** [comment]
@stuartmorgan installing a newer version typically grabs the latest windows SDK by default hence the need to either manually add the compatible older one in VS Installer or perform a project upgrade as prompted by VS if the older SDK isn’t there.  Regarding the build tools, not sure if 2019 installs v141 and v142 side by side by default.  Also not aware of a “use this one or greater” option.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Saifallak** [comment]
@clarkezone 2019 installs v142 by default and manually you select to install v141

**Comment by AppleEducate** [comment]
On the latest VSCode and MacOS the flutter desktop app does not show anything in the debug console, is that because of the flutter tool or the desktop embedding?

**Comment by stuartmorgan** [comment]
That's a Flutter tool bug that was fixed yesterday. It has nothing to do with this issue, which is about compiling the Windows shell using Visual Studio 2019.

**Comment by AppleEducate** [comment]
Perfect! Wasn’t sure it it was specially VsCode

**Comment by mtcliatt** [comment]
I installed VS 2017, which by default installed the newer Windows 10 SDK version, `10.0.17763.0`. To get the compatible older SDK version (`10.0.17134.0`), I had to manually select it from the VS installer. 

It sounds like this is the case for both VS 2017 and VS 2019 (as per MichaelPriebe's comments above).

**Comment by stuartmorgan** [comment]
I have [a branch](https://github.com/stuartmorgan/flutter-desktop-embedding/tree/windows-sdk-update) that I believe will fix this for both 2017 and 2019 without extra installs, but as I currently have VS 2017 I'll need someone with VS 2109 to test it out to make sure it's working correctly.
- It unconditionally updates the SDK to 10.0.17763.0. This part I know works.
- It *should* make the toolchain switch based on VS version. The condition format is documented, but I've never used it so may have made a mistake.

If someone with VS 2019 could check out that branch and check that:
- If you inspect the Properties for the Runner project, `Platform Toolset` shows v142/Visual Studio 2019
- If you build with that toolset, it does actually work (e.g., there are no new errors flagged by v142 that didn't happen under v141)

then report back here, that would be very helpful.

**Comment by contrix09** [comment]
@stuartmorgan I am running VS2019 and trying to build the project from your branch. But I am having this error.

```
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(128,5): error MSB3073: The command "C:\Users\CADN\source\repos\Flutter Demo App\example\windows\scripts\prepare_dependencies "C:\Users\CADN\source\repos\Flutter Demo App\example\windows\..\build\windows\intermediates\x64\Debug\Runner\"
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(128,5): error MSB3073: :VCEnd" exited with code 9009.
1>Done building project "Runner.vcxproj" -- FAILED.
```
Although the `Platform Toolset` in the properties already shows `VS2019 (v142)`:

![image](https://user-images.githubusercontent.com/29834445/57816348-8ff8b700-77ad-11e9-9e58-c5300d581efa.png)

**Comment by stuartmorgan** [comment]
I think you may have had exceptionally poor luck in when you pulled, as a change to the Windows build process that involved both a Flutter side patch and an FDE-side patch *just* landed, and there was a small window where if you updated Flutter to the latest version the build would be broken.

Could you try updating both Flutter and your pull of the branch (I've merged in the FDE change) and try again?

**Comment by contrix09** [comment]
I updated Flutter and pulled your changes, but the error has changed to this:

```
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v150\Microsoft.CppCommon.targets(128,5): error MSB3073: The command "C:\Users\CADN\source\repos\Flutter Demo App\flutter_desktop_embedding\example\windows\scripts\prepare_dependencies debug
1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v150\Microsoft.CppCommon.targets(128,5): error MSB3073: :VCEnd" exited with code 9009.
1>Done building project "Runner.vcxproj" -- FAILED.
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========
```

**Comment by stuartmorgan** [comment]
That build script could definitely use more logging. Is this from within VS, or via `flutter run` (or `flutter build`)? You'll need to do the first build with the `flutter` tool after updating.

**Comment by contrix09** [comment]
That was from within VS. I tried to run the command `flutter build windows -v` but shows me an error:

```
C:\Users\CADN\source\repos\Flutter Demo App\flutter_desktop_embedding\example>flutter build windows -v
[  +30 ms] executing: [C:\Users\CADN\.flutter\flutter\] git log -n 1 --pretty=format:%H
[  +74 ms] Exit code 0 from: git log -n 1 --pretty=format:%H
[   +1 ms] 4e1bfca8470ff088ecb68aaa72086f290b316fd1
[   +1 ms] executing: [C:\Users\CADN\.flutter\flutter\] git describe --match v*.*.* --first-parent --long --tags
[  +83 ms] Exit code 0 from: git describe --match v*.*.* --first-parent --long --tags
[        ] v1.5.8-267-g4e1bfca84
[  +11 ms] executing: [C:\Users\CADN\.flutter\flutter\] git rev-parse --abbrev-ref --symbolic @{u}
[  +54 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[   +1 ms] origin/master
[   +1 ms] executing: [C:\Users\CADN\.flutter\flutter\] git ls-remote --get-url origin
[  +47 ms] Exit code 0 from: git ls-remote --get-url origin
[   +1 ms] https://github.com/flutter/flutter.git
[  +80 ms] executing: [C:\Users\CADN\.flutter\flutter\] git rev-parse --abbrev-ref HEAD
[  +53 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[   +1 ms] master
[  +42 ms] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[  +10 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[   +9 ms] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[ +107 ms] Building Windows application...
[  +24 ms] Building Windows application... (completed in 0.0s)
[   +4 ms] "flutter windows" took 259ms.
[   +1 ms] "flutter windows" took 259ms.
Build process failed

#0      throwToolExit (package:flutter_tools/src/base/common.dart:24:3)
#1      buildWindows (package:flutter_tools/src/windows/build_windows.dart:56:5)
<asynchronous suspension>
#2      BuildWindowsCommand.runCommand (package:flutter_tools/src/commands/build_windows.dart:62:11)
<asynchronous suspension>
#3      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:488:18)
#4      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:77:64)
#5      _rootRunUnary (dart:async/zone.dart:1132:38)
#6      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#7      _FutureListener.handleValue (dart:async/future_impl.dart:126:18)
#8      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:639:45)
#9      Future._propagateToListeners (dart:async/future_impl.dart:668:32)
#10     Future._complete (dart:async/future_impl.dart:473:7)
#11     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#12     _AsyncAwaitCompleter.complete.<anonymous closure> (dart:async-patch/async_patch.dart:33:20)
#13     _rootRun (dart:async/zone.dart:1124:13)
#14     _CustomZone.run (dart:async/zone.dart:1021:19)
#15     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:947:23)
#16     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#17     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#18     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:115:13)
#19     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:5)



[  +13 ms] 'C:\Program' is not recognized as an internal or external command,
[   +3 ms] operable program or batch file.
```

Maybe I'm missing something here?

**Comment by stuartmorgan** [comment]
That's a concerning error, but more fundamental than the VS 2019 issue here. I've filed 32792 to track that down separately.

**Comment by stuartmorgan** [comment]
I'm going to go ahead and land that branch; since the conditional is working, that's the right approach regardless. If anyone hits issues compiling with the v142 toolchain, please file new bugs and they can be fixed in follow-ups.

**Comment by stuartmorgan** [comment]
I forgot to tag the issue in the PR; fixed by #385.
